src/components/Contacts/ContactDetails/ContactDetailsHeader/DeleteContactModal/DeleteContactModal.tsx

```diff
@@ -12,37 +17,178 @@ import {
   DeleteButton,
 } from 'src/components/common/Modal/ActionButtons/ActionButtons';
 import Modal from 'src/components/common/Modal/Modal';
+import { useAccountListId } from 'src/hooks/useAccountListId';
+import { isEditableSource, sourceToStr } from 'src/utils/sourceHelper';
+import { useContactSourceQuery } from './ContactSource.generated';
 
 const LoadingIndicator = styled(CircularProgress)(({ theme }) => ({
   margin: theme.spacing(0, 1, 0, 0),
 }));
 
+interface DataInfo {
+  canDeleteWithoutIssues: boolean;
+  contactSource: string;
+  addressSources: string[];
+  emailSources: string[];
+  phoneSources: string[];
+}
 interface DeleteContactModalProps {
   open: boolean;
   setOpen: (open: boolean) => void;
   deleting: boolean;
   deleteContact: () => void;
+  contactId: string;
 }
 
 export const DeleteContactModal: React.FC<DeleteContactModalProps> = ({
   open,
   setOpen,
   deleting,
   deleteContact,
+  contactId,
 }) => {
   const { t } = useTranslation();
+  const accountListId = useAccountListId() ?? '';
+
+  const { data } = useContactSourceQuery({
+    variables: { accountListId, contactId },
+    skip: !open && !contactId,
+  });
+  const contactSources = data?.contact;
+
+  const dataInfo: DataInfo = useMemo(() => {
+    if (!contactSources) {
+      return {
+        canDeleteWithoutIssues: true,
+        contactSource: sourceToStr(t, 'MPDX'),
+        addressSources: [],
+        emailSources: [],
+        phoneSources: [],
+      };
+    }
+
+    // We ensure the contact was created on MPDX and that all the data is editable.
+    // If any data is not editable, this means it was created by a third party.
+    // Which will only recreate the data after deleting it on MPDX.
+    // To prevent this confusion, we do not allow a contact to be deleted if it has non editable data.
+
+    const addressSources = new Set<string>();
+    const emailSources = new Set<string>();
+    const phoneSources = new Set<string>();
+
+    contactSources.addresses?.nodes.forEach((address) => {
+      if (!isEditableSource(address.source)) {
+        addressSources.add(address.source);
+      }
+    });
+
+    contactSources.people?.nodes?.forEach((person) => {
+      person.emailAddresses.nodes.forEach((email) => {
+        if (!isEditableSource(email.source)) {
+          emailSources.add(email.source);
+        }
+      });
+      person.phoneNumbers.nodes.forEach((phone) => {
+        if (!isEditableSource(phone.source)) {
+          phoneSources.add(phone.source);
+        }
+      });
+    });
+
+    return {
+      canDeleteWithoutIssues:
+        isEditableSource(contactSources.source ?? '') &&
```

Comment:
If the source is an empty string, `isEditableSource` will return false. I think we want to convert `null` to `undefined` so that `isEditableSource` returns true.

```suggestion
        isEditableSource(contactSources.source ?? undefined) &&
```